spec.bs

```diff
@@ -8349,6 +8628,97 @@ scoring signals requests=] |entriesToBatch|:
+  of (interger, integer).
+1. Let |nextCompressionGroupId| be 0.
+1. [=map/For each=] |request| of |entriesToBatch|:
+  1. Let |mapKey| be a tuple of (|request|'s [=trusted scoring signals request/owner origin=],
+    |request|'s [=trusted scoring signals request/joining origin=]).
+  1. If |compressionIdMap| does not [=map/contain=] |mapKey|:
+    1. Set |compressionIdMap|[|mapKey|] to |nextCompressionGroupId|.
+    1. Increment |nextCompressionGroupId| by 1.
+  1. Let |compressionGroupId| be |compressionIdMap|[|mapKey|].
+  1. If |compressionGroupMap| does not [=map/contain=] |compressionGroupId|:
+    1. Set |compressionGroupMap|[|compressionGroupId|] to an empty [=map=], whose [=map/keys=] are integers
+  1. Let |partitionId| be [=list/size=] of |compressionGroupMap|[|compressionGroupId|].
+  1. Set |renderUrlIdMap|[|request|'s [=trusted scoring signals request/render URL=]] to [=tuple=]
+    of (|compressionGroupId|, |partitionId|).
+  1. Let |partition| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
+      [=list=] of [=strings=].
+  1. Set |partition|["id"] to |partitionId|.
+  1. Let |namespace| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
+      [=list=] of [=strings=].
+  1. Set |namespace|["render_url"] to [|request|'s [=trusted scoring signals request/render URL=]].
+  1. Set |namespace|["ad_component_render_urls"] to |request|'s [=trusted scoring signals request/ad
+    component URLs=].
+  1. Set |partition|["namespace"] to |namespace|.
+  1. Let |partitionMetadata| be an empty [=map=], whose [=map/keys=] and [=map/values=] are [=strings=].
+  1. Set |partitionMetadata|["experiment_group_id"] to |firstRequest|'s [=trusted scoring signals
+      request/seller experiment group id=].
+  1. Set |partition|["metadata"] to |partitionMetadata|.
+  1. Set |compressionGroupMap|[|compressionGroupId|][|partitionId|] to |partition|.
+1. [=map/For each=] |id| → |group| in |compressionGroupMap|:
+  1. Let |compressionGroup| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are integers or
+    [=lists=] of [=maps=].
+  1. Set |compressionGroup|["compression_group_id"] to |id|.
+  1. Set |compressionGroup|["partitions"] to an empty [=list=].
+  1. [=list/For each=] |partition| in |group|'s [=map/values=]:
+    1. [=list/Append=] |partition| to |compressionGroup|["partitions"].
+  1. [=list/Append=] |compressionGroup| to |compressionGroups|.
+1. Let |metadata| be an empty [=map=], whose [=map/keys=] and [=map/values=] are [=strings=].
+1. Let |hostname| be the result of [=string/UTF-8 percent-encoding=] the
+  [=serialization of an origin|serialized=] |firstRequest|'s [=trusted scoring signals request/top
+    level origin=] using [=component percent-encode set=].
+1. Set |metadata|["hostname"] to |hostname|.
+1. Let |keyInfo| be the result of [=looking up the server encryption key=] with |firstRequest|'s

```

Comment:
same here. Look up algorithm can return failure. I think we need to handle that.

Reply:
Change to the same handler as bidding signals.